src/days-2-4-adv/adv.py

```diff
@@ -49,3 +49,59 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+while True:
+    print(f'\nLocation: {player.location.name}')
+    if hasattr(player.location, 'visibility'):
+        for item in player.inventory:
+            if hasattr(item, 'light'):
+                print(f"The {item.name} gives light to the dark room!")
+                player.location.light_on()
+                print(f'{player.location.description}')
+                if len(player.location.items) > 0:
+                    player.location.print_items()
+        if player.location.visibility is not True:
+            print(f"The room is dark and you can't see anything!")
+    else:
+        print(f'{player.location.description}')
+        if len(player.location.items) > 0:
+            player.location.print_items()
+
+    player_input = input(f'\nWhat does {player.name} do?   ')
+    player_input_args = player_input.split(' ')
+
+    if len(player_input) < 1:
+        print('\nControls: \nN,S,E,W to move to a different room. \nTake ___ to take an item. \nDrop ___ to drop an item. \nI to view your inventory. \nZ to view score \nQ to quit')
+
+    if len(player_input) == 1:
+        if player_input[0].lower() == "q":
+            break
+        elif player_input[0].lower() == "n" or player_input[0].lower() == "s" or player_input[0].lower() == "w" or player_input[0].lower() == "e":
+            player.change_location(player_input[0])
+        elif player_input[0].lower() == "i":
+            player.view_inventory()
+        elif player_input[0].lower() == "z":
+            player.get_score()
+        else:
+            print('\nControls: \nN,S,E,W to move to a different room. \nTake ___ to take an item. \nDrop ___ to drop an item. \nI to view your inventory. \nZ to view score \nQ to quit')
+
+    if len(player_input_args) == 2:
+        if player_input[0].lower() == "t" or player_input[0].lower() == "take":
+            taking_item = player.location.find_item(player_input_args[1])
+            if taking_item == None:
+                print(f"You couldn't find a {player_input_args[1]}")
+            else:
+                if hasattr(taking_item, 'value'):
+                    player.add_to_score(taking_item.value)
+                    if taking_item.value > 0:
+                        print(f"You gained {taking_item.value} points for finding the {taking_item.name}")
+                    taking_item.remove_value()
+                player.take_item(taking_item)
+                player.location.remove_item(taking_item)
+
+        elif player_input[0].lower() == "d" or player_input[0].lower() == "drop":
+            dropped_item = player.find_item(player_input_args[1])
+            if dropped_item.name == None:
```

Comment:
Hmm. I am getting a NoneType error message saying object has no attribute "name". Maybe this is referring to the find_item method?? Not 100% sure.